sound/soc/sof/sof-client-ipc-msg-injector.c

```diff
@@ -221,27 +224,65 @@ static ssize_t sof_msg_inject_ipc4_dfs_write(struct file *file,
 	return count;
 };
 
+static ssize_t sof_kernel_msg_inject_dfs_write(struct file *file, const char __user *buffer,
+					       size_t count, loff_t *ppos)
+{
+	struct sof_client_dev *cdev = file->private_data;
+	struct snd_sof_dev *sdev = cdev->sdev;
```

Comment:
SOF client drivers must never use 'sdev'

Reply:
`sdev` is used indirectly via `sof_client_ipc_tx_message` would you rather I shove the callback logic into that file?

Reply:
The `sof_client_ipc_tx_message()` is within snd_sof.ko it can use sdev, the client drivers are _not_ within the core SOF modules and they must not.